src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/git-clone-to-dir.ps1

```
@@ -0,0 +1,111 @@
+
+  if ($CopyWip) {
+    # Copy over changes that haven't been committed, for dev inner loop.
+    # This gets changes (whether staged or not) but misses untracked files.
+    $stashCommit = (git stash create)
+
+    if ($stashCommit) {
+      Write-Host "WIP changes detected: created temporary stash $stashCommit to transfer to inner repository..."
+    }
+    else {
+      Write-Host "No WIP changes detected..."
+    }
+  }
+
+  Write-Host "Creating empty clone at: $Dest"
+
+  # Combine the results of git rev-parse --git-dir with the "shallow" subdirectory
+  # to get the full path to the shallow file.
+
+  $shallowFile = Join-Path (git rev-parse --git-dir) "shallow"
+
+  if (Test-Path $shallowFile) {
+    Write-Host "Source repository is shallow..."
+    if ($stashCommit) {
+      Write-Host "WIP stash is not supported in a shallow repository: aborting."
+      exit 1
+    }
+
+    # If source repo is shallow, old versions of Git refuse to clone it to another directory. First,
+    # remove the 'shallow' file to trick Git into allowing the clone.
+    $shallowContent = Get-Content -Path $shallowFile
+    Remove-Item -Path $shallowFile
+
+    # Then, run the clone:
+      # * 'depth=1' avoids encountering the leaf commit in the shallow repo that points to a parent
+      #   that doesn't exist. (The commit marked "grafted".) Git would fail here, otherwise.
+      # * '--no-local' allows a shallow clone from a git dir on the same filesystem. This means the
+      #   clone will not use hard links and takes up more space. However, since we're doing a shallow
+      #   clone anyway, the difference is probably not significant. (This has not been measured.)
+      # * '-c protocol.file.allow=alwyas allows cloning from the local filesystem even on newer versions
+      #   of git which have disabled this for security reasons.
+    git clone -c protocol.file.allow=always --depth=1 --no-local --no-checkout "$Source" "$Dest"
```

When native executables calls fail, the error is not propagated in PowerShell (unless I'm missing some external setup) and $LASTEXITCODE has to be checked. Do we have a solution for this in arcade (such as a wrapper function for calling native binaries that checks the LASTEXITCODE and throws if non-zero) or that's something that we'd need to do yet?